tests/test_monitor.py

```diff
@@ -0,0 +1,116 @@
+"""Unit tests for monitor module."""
+
+import unittest
+from unittest import mock
+from typing import Optional, Tuple
+
+from PyQt5.QtWidgets import QApplication
+
+from iquip import monitor
+
+class TestMonitor(unittest.TestCase):
+    """Unit tests for Monitor class."""
+
+    def test_init_initial_value(self):
+        for value in (True, None, 1.0, "value", object()):
+            mon = monitor.Monitor(initial_value=value)
+            self.assertEqual(mon._value, value)
+
+    def test_init_updated_callback(self):
+        callback = mock.MagicMock()
+        mon = monitor.Monitor(initial_value=None, updated_callback=callback)
+        mon.updated_callback(None)
+        callback.assert_called_once_with(None)
+
+    def test_get_value(self):
+        for value in (True, None, 1.0, "value", object()):
+            mon = monitor.Monitor(initial_value=value)
+            self.assertEqual(mon.get_value(), mon._value)
+
+    def test_set_value(self):
+        values = ((True, False), (1, 0), ("old", "new"))
+        for init_value, new_value in values:
+            mon = monitor.Monitor(initial_value=init_value)
+            mon.set_value(new_value)
+            self.assertEqual(mon._value, new_value)
+
+    def test_set_value_callback(self):
+        callback = mock.MagicMock()
+        mon = monitor.Monitor(initial_value="old", updated_callback=callback)
+        mon.set_value("new")
+        callback.assert_called_once_with("new")
+
+    def test_update(self):
+        mon = monitor.Monitor(initial_value="old")
+        with mock.patch.multiple(mon, set_value=mock.DEFAULT, _read=mock.DEFAULT) as mocked:
+            mocked["_read"].return_value = "new"
+            mon.update()
+        mocked["set_value"].assert_called_once_with("new")
```

Comment:
How about moving this line into `with` statement?
I think it is better because `mocked` is defined in `with` block.

Reply:
Actually I am still not sure about this.
As you said, it is defined in the `with` block so it seems intuitive to use it in the block.
However, it is in fact not a problem in Python's scope system.
The reason why I put it outside the `with` block is that the _patching_ is not required any more at that point.
I kept the same policy for other patching codes.

How do you think of this?

Reply:
More of my opinion:

According to [PEP343](https://peps.python.org/pep-0343/#specification-the-with-statement), a `with` statement is actually a special form of `try-except-finally` statements.
In Python, variables defined in `try` block can be used outside and it is quite common as far as I know.
So I think it could be accepted to use `mocked` outside the `with` block.
Since a `with` block is a _context-managing_ area, the more important point is "until when the context should be maintained".

Reply:
Wow, thanks for the reference!
As you said, there is no problem for Python's scope system and I think it's better to leave it as it is.